### PR TITLE
416 resolve light warnings for mylyn gitlab connector

### PR DESCRIPTION
**Pull request by gnl42**
A lot of strings than can probably be combined/shared

### CODE REVIEW

File: mylyn.tasks/connectors/gitlab/org.eclipse.mylyn.gitlab.core/src/org/eclipse/mylyn/internal/gitlab/core/GitlabOperation.java [hunk 1]
```diff
@@ -99,10 +100,10 @@ protected void validate(CommonHttpResponse response, int expected, IOperationMon
 		if (statusCode != expected /* && statusCode != HttpStatus.SC_BAD_REQUEST */) {
 			if (statusCode == HttpStatus.SC_NOT_FOUND) {
 				throw new GitlabException(new Status(IStatus.ERROR, GitlabCoreActivator.PLUGIN_ID,
-						NLS.bind("Requested resource ''{0}'' does not exist", response.getRequestPath())));
+						NLS.bind("Requested resource ''{0}'' does not exist", response.getRequestPath()))); //$NON-NLS-1$
```
Comment (ruspl-afed) [inline mylyn.tasks/connectors/gitlab/org.eclipse.mylyn.gitlab.core/src/org/eclipse/mylyn/internal/gitlab/core/GitlabOperation.java:103]: this should i18ned

File: mylyn.tasks/connectors/gitlab/org.eclipse.mylyn.gitlab.core/src/org/eclipse/mylyn/internal/gitlab/core/GitlabOperation.java [hunk 2]
```diff
@@ -99,10 +100,10 @@ protected void validate(CommonHttpResponse response, int expected, IOperationMon
...
 			}
 			throw new GitlabException(new Status(IStatus.ERROR, GitlabCoreActivator.PLUGIN_ID,
-					NLS.bind("Unexpected response from Gitlab REST server for ''{0}'': {1}", response.getRequestPath(),
+					NLS.bind("Unexpected response from Gitlab REST server for ''{0}'': {1}", response.getRequestPath(), //$NON-NLS-1$
```
Comment (ruspl-afed) [inline mylyn.tasks/connectors/gitlab/org.eclipse.mylyn.gitlab.core/src/org/eclipse/mylyn/internal/gitlab/core/GitlabOperation.java:106]: this should i18ned

File: mylyn.tasks/connectors/gitlab/org.eclipse.mylyn.gitlab.core/src/org/eclipse/mylyn/internal/gitlab/core/GitlabOperation.java [hunk 3]
```diff
@@ -52,7 +53,7 @@ public T run(IOperationMonitor monitor) throws GitlabException {
 			return execute(monitor);
 		} catch (IOException e) {
 			throw new GitlabException(new Status(IStatus.ERROR, GitlabCoreActivator.PLUGIN_ID,
-					"org.eclipse.mylyn.gitlab.core.GitlabOperation.run(IOperationMonitor)", e));
+					"org.eclipse.mylyn.gitlab.core.GitlabOperation.run(IOperationMonitor)", e)); //$NON-NLS-1$
```
Comment (ruspl-afed) [inline mylyn.tasks/connectors/gitlab/org.eclipse.mylyn.gitlab.core/src/org/eclipse/mylyn/internal/gitlab/core/GitlabOperation.java:56]: this message should be improved and i18ned

File: mylyn.tasks/connectors/gitlab/org.eclipse.mylyn.gitlab.core/src/org/eclipse/mylyn/internal/gitlab/core/GitlabPutOperation.java [hunk 4]
```diff
@@ -47,7 +48,7 @@ protected void addHttpRequestEntities(HttpRequestBase request) throws GitlabExce
 			((HttpPut) request).setEntity(new StringEntity(body));
 		} catch (UnsupportedEncodingException e) {
 			throw new GitlabException(
-					new Status(IStatus.ERROR, GitlabCoreActivator.PLUGIN_ID, "UnsupportedEncodingException", e));
+					new Status(IStatus.ERROR, GitlabCoreActivator.PLUGIN_ID, "UnsupportedEncodingException", e)); //$NON-NLS-1$
```
Comment (ruspl-afed) [inline mylyn.tasks/connectors/gitlab/org.eclipse.mylyn.gitlab.core/src/org/eclipse/mylyn/internal/gitlab/core/GitlabPutOperation.java:51]: this message should be improved and i18ned

File: mylyn.tasks/connectors/gitlab/org.eclipse.mylyn.gitlab.core/src/org/eclipse/mylyn/internal/gitlab/core/GitlabRepositoryConnector.java [hunk 5]
```diff
@@ -241,7 +241,7 @@ public String getConnectorKind() {
 
 	@Override
 	public String getLabel() {
-		return "Gitlab";
+		return "Gitlab"; //$NON-NLS-1$
```
Comment (ruspl-afed) [inline mylyn.tasks/connectors/gitlab/org.eclipse.mylyn.gitlab.core/src/org/eclipse/mylyn/internal/gitlab/core/GitlabRepositoryConnector.java:244]: this should be i18ned

File: mylyn.tasks/connectors/gitlab/org.eclipse.mylyn.gitlab.core/src/org/eclipse/mylyn/internal/gitlab/core/GitlabRepositoryConnector.java [hunk 6]
```diff
@@ -276,15 +276,15 @@ public TaskData getTaskData(TaskRepository repository, String taskId, IProgressM
 	@Override
 	public IStatus performQuery(TaskRepository repository, IRepositoryQuery query, TaskDataCollector collector,
 			ISynchronizationSession session, IProgressMonitor monitor) {
-		monitor.beginTask("performQuery", IProgressMonitor.UNKNOWN);
+		monitor.beginTask("performQuery", IProgressMonitor.UNKNOWN); //$NON-NLS-1$
```
Comment (ruspl-afed) [inline mylyn.tasks/connectors/gitlab/org.eclipse.mylyn.gitlab.core/src/org/eclipse/mylyn/internal/gitlab/core/GitlabRepositoryConnector.java:279]: this message should be i18ned

File: mylyn.tasks/connectors/gitlab/org.eclipse.mylyn.gitlab.core/src/org/eclipse/mylyn/internal/gitlab/core/GitlabRepositoryConnector.java [hunk 7]
```diff
@@ -276,15 +276,15 @@ public TaskData getTaskData(TaskRepository repository, String taskId, IProgressM
...
 		GitlabRestClient client;
 		try {
 			client = getClient(repository);
 			IOperationMonitor progress = OperationUtil.convert(monitor, "performQuery", 3); //$NON-NLS-1$
 			client.getIssues(query, collector, new NullOperationMonitor());
 		} catch (CoreException e) {
 			return new Status(IStatus.ERROR, GitlabCoreActivator.PLUGIN_ID, IStatus.INFO,
-					"CoreException from performQuery", e);
+					"CoreException from performQuery", e); //$NON-NLS-1$
```
Comment (ruspl-afed) [inline mylyn.tasks/connectors/gitlab/org.eclipse.mylyn.gitlab.core/src/org/eclipse/mylyn/internal/gitlab/core/GitlabRepositoryConnector.java:287]: this message should be improved and i18ned

File: mylyn.tasks/connectors/gitlab/org.eclipse.mylyn.gitlab.core/src/org/eclipse/mylyn/internal/gitlab/core/GitlabRepositoryConnector.java [hunk 8]
```diff
@@ -306,7 +306,7 @@ public RepositoryInfo validateRepository(TaskRepository repository, IProgressMon
 			GitlabRestClient client = createClient(repository);
 			if (!client.validate(OperationUtil.convert(monitor))) {
 				throw new CoreException(
-						new Status(IStatus.ERROR, GitlabCoreActivator.PLUGIN_ID, "repository is invalide"));
+						new Status(IStatus.ERROR, GitlabCoreActivator.PLUGIN_ID, "repository is invalide")); //$NON-NLS-1$
```
Comment (ruspl-afed) [inline mylyn.tasks/connectors/gitlab/org.eclipse.mylyn.gitlab.core/src/org/eclipse/mylyn/internal/gitlab/core/GitlabRepositoryConnector.java:309]: this message should be i18ned

File: mylyn.tasks/connectors/gitlab/org.eclipse.mylyn.gitlab.ui/src/org/eclipse/mylyn/internal/gitlab/ui/TaskEditorActivityPart.java [hunk 9]
```diff
@@ -189,7 +190,7 @@ private void createActivityList(FormToolkit toolkit) {
 								// no character under event.x, event.y
 							}
 							if (style1 != null && style1.underline && style1.underlineStyle == SWT.UNDERLINE_LINK) {
-								System.out.println("Click on a Link " + style1.data);
+								System.out.println("Click on a Link " + style1.data); //$NON-NLS-1$
```
Comment (ruspl-afed) [inline mylyn.tasks/connectors/gitlab/org.eclipse.mylyn.gitlab.ui/src/org/eclipse/mylyn/internal/gitlab/ui/TaskEditorActivityPart.java:193]: this statement should be removed

File: mylyn.tasks/connectors/gitlab/org.eclipse.mylyn.gitlab.ui/src/org/eclipse/mylyn/internal/gitlab/ui/TaskEditorActivityPart.java [hunk 10]
```diff
@@ -206,8 +207,8 @@ private void createActivityList(FormToolkit toolkit) {
 						}
 						if (style1 != null && style1.underline && style1.underlineStyle == SWT.UNDERLINE_LINK) {
 							String oldTooltip = text.getToolTipText();
-							text.setToolTipText("" + style1.data);
-							Job job = new Job("Deactivate Tooltip") {
+							text.setToolTipText("" + style1.data); //$NON-NLS-1$
+							Job job = new Job("Deactivate Tooltip") { //$NON-NLS-1$
```
Comment (ruspl-afed) [inline mylyn.tasks/connectors/gitlab/org.eclipse.mylyn.gitlab.ui/src/org/eclipse/mylyn/internal/gitlab/ui/TaskEditorActivityPart.java:211]: this message should be i18ned

File: mylyn.tasks/connectors/gitlab/org.eclipse.mylyn.gitlab.ui/src/org/eclipse/mylyn/internal/gitlab/ui/TaskEditorActivityPart.java [hunk 11]
```diff
@@ -119,7 +120,7 @@ private void expandSection(FormToolkit toolkit, Section section) {
 		if (activityAttributes.size() > 0) {
 			createActivityList(toolkit);
 		} else {
-			toolkit.createLabel(activityComposite, "No Activity Events");
+			toolkit.createLabel(activityComposite, "No Activity Events"); //$NON-NLS-1$
```
Comment (ruspl-afed) [inline mylyn.tasks/connectors/gitlab/org.eclipse.mylyn.gitlab.ui/src/org/eclipse/mylyn/internal/gitlab/ui/TaskEditorActivityPart.java:123]: this message should be i18ned

File: mylyn.tasks/connectors/gitlab/org.eclipse.mylyn.gitlab.ui/src/org/eclipse/mylyn/internal/gitlab/ui/TaskEditorActivityPart.java [hunk 12]
```diff
@@ -63,7 +64,7 @@ public class TaskEditorActivityPart extends AbstractTaskEditorPart {
 	private final Gson gson = new Gson();
 
 	public TaskEditorActivityPart() {
-		setPartName("Activity Events");
+		setPartName("Activity Events"); //$NON-NLS-1$
```
Comment (ruspl-afed) [inline mylyn.tasks/connectors/gitlab/org.eclipse.mylyn.gitlab.ui/src/org/eclipse/mylyn/internal/gitlab/ui/TaskEditorActivityPart.java:67]: this message should be i18ned

**Review by ruspl-afed — CHANGES_REQUESTED** [review]
Adding //NON-NLS marker should be done only for strings that do not need translation.
But here I can see  //NON-NLS added for all strings blindly, with masking strings that needs translation.

The rule is quite simple: everything that _can_ appear in UI (GUI, CLI, whatever else) should be translated.

Please rework this PR fully (I stopped to mark problematic places when realized the applied approach), we don't need to suppress warnings "at all cost", we need to resolve them.

**Comment by gnl42** [comment]
That goes beyond light warning, more like a refactor.

I don't know what the GitLab connector should look like so I'm cancelling the PR and deleting the branch

**Pull request closed without merging**